Develop/scripts/script.js

```diff
@@ -0,0 +1,198 @@
+$(document).ready(function () {
+  var cityArray = [];
+  var lon = "";
+  var lat = "";
+  var apiKey = "39edc21fee70b725a0dbcb6dbed85366";
+  var city = "";
+
+  retrieveCityArray();
+
+  // When user searches for city
+  $("#add-city").on("click", function (event) {
+    event.preventDefault();
+
+    storeCityArray();
+    retrieveCityArray();
+  });
+
+  // When user clicks city button
+  $(document).on("click", ".city-button", getCityClicked);
+
+  function storeCityArray() {
+    city = $("#city-input").val().trim();
```

Comment:
You want to aim for pure functions. Your function shouldn't be mutating a "global" application variable like city. Pass values into a function, work out a new value, return that, store that etc. If your function is effecting variables outside of its scope it can't be pure.